garak/generators/base.py

```diff
@@ -124,7 +124,15 @@ def generate(
                     for result in pool.imap_unordered(
                         self._call_model, [prompt] * generations_this_call
                     ):
-                        outputs.append(result)
+                        if len(result) != 1:
+                            message = f"Expected a list of length 1 for generator with support_multiple_generations=False, got {len(result)}"
+                            logging.error(message)
+                            raise ValueError(message)
+                        if not isinstance(result[0], str):
+                            message = f"Expected string result got {type(result[0])}"
+                            logging.error(message)
+                            raise ValueError(message)
+                        outputs.append(result[0])
```

Comment:
This does not match the expectations for `_call_model()` which allows a return value of `List[Union[str, None]]`. It should be valid for `result[0] == None`. The clause should be the same as used in the single result case:
https://github.com/leondz/garak/blob/bebd8b39e37cba1081ecfe40217fb724d159831a/garak/generators/base.py#L148

Going one step further this code has a common function to:
https://github.com/leondz/garak/blob/bebd8b39e37cba1081ecfe40217fb724d159831a/garak/generators/base.py#L141-L150

It might be more maintainable to add a method such as:
``` Python
    @staticmethod
    def _verify_model_result(result):
        assert isinstance(
            result, list
        ), "_call_model must return a list"
        assert (
                len(result) == 1
        ), "_call_model must return a list of one item when invoked as _call_model(prompt, 1)"
        assert (
            isinstance(result[0], str) or result[0] is None
        ), "_call_model's item must be a string or None"
```

``` Suggestion
                        _verify_model_result(result)
                        outputs.append(result[0])
```

And update the serial case to:
``` Python
                    _verify_model_result(output_one)
                    outputs.append(output_one[0])
```

Reply:
Thanks for catching this @jmartin-tech , I've updated the pull request with your suggestions in https://github.com/leondz/garak/pull/864/commits/3551a348a08ef8f77054c53a0fdce617e8cbbddd